ImageAnalysis/image_analysis/offline_analyzers/HASO_himg_has_processor.py

```diff
@@ -0,0 +1,379 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional, Union, NamedTuple
+from pathlib import Path
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
+    from image_analysis.types import Array2D
+
+try:
+    import image_analysis.third_party_sdks.wavekit_43.wavekit_py as wkpy
+except ModuleNotFoundError as e:
+    errmsg = "could not import wkpy, e.g. might be running on non windows machine"
+    e.add_note(errmsg)
```

Comment:
`add_note` was added in Python 3.11, whereas ImageAnalysis should support 3.7

```suggestion
    e.args += (errmsg,)
```